src/frontend/src/functions/auth.tsx

```diff
@@ -1,60 +1,62 @@
 import { t } from '@lingui/macro';
-import { notifications, showNotification } from '@mantine/notifications';
+import { notifications } from '@mantine/notifications';
 import { IconCheck } from '@tabler/icons-react';
 import axios from 'axios';
 
 import { api } from '../App';
 import { ApiEndpoints } from '../enums/ApiEndpoints';
-import { apiUrl, useServerApiState } from '../states/ApiState';
+import { apiUrl } from '../states/ApiState';
 import { useLocalState } from '../states/LocalState';
 import { useSessionState } from '../states/SessionState';
-import {
-  useGlobalSettingsState,
-  useUserSettingsState
-} from '../states/SettingsState';
-import { useUserState } from '../states/UserState';
 
-export const doClassicLogin = async (username: string, password: string) => {
+/**
+ * Attempt to login using username:password combination.
+ * If login is successful, an API token will be returned.
+ * This API token is used for any future API requests.
+ */
+export const doBasicLogin = async (username: string, password: string) => {
   const { host } = useLocalState.getState();
+  // const apiState = useServerApiState.getState();
+
+  if (username.length == 0 || password.length == 0) {
+    return;
+  }
 
-  // Get token from server
-  const token = await axios
+  // At this stage, we can assume that we are not logged in, and we have no token
+  useSessionState.getState().clearToken();
+
+  // Request new token from the server
+  await axios
     .get(apiUrl(ApiEndpoints.user_token), {
       auth: { username, password },
       baseURL: host,
-      timeout: 2000,
+      timeout: 2500,
       params: {
         name: 'inventree-web-app'
       }
     })
-    .then((response) => response.data.token)
-    .catch((error) => {
-      showNotification({
-        title: t`Login failed`,
-        message: t`Error fetching token from server.`,
-        color: 'red'
-      });
-      return false;
-    });
-
-  if (token === false) return token;
-
-  // log in with token
-  doTokenLogin(token);
-  return true;
+    .then((response) => {
+      if (response.status == 200 && response.data.token) {
+        // A valid token has been returned - save, and login
+        useSessionState.getState().setToken(response.data.token);
+      }
+    })
+    .catch(() => {});
 };
 
 /**
- * Logout the user (invalidate auth token)
+ * Logout the user from the current session
+ *
+ * @arg deleteToken: If true, delete the token from the server
  */
-export const doClassicLogout = async () => {
-  // Set token in context
-  const { setToken } = useSessionState.getState();
-
-  setToken(undefined);
+export const doClassicLogout = async (navigate: any, deleteToken?: boolean) => {
```

Comment:
Comment: The other auth method was renamed so maybe this should also be renamed to be consitant

Reply:
I have renamed this now.